packages/dev/loaders/src/SPLAT/splatFileLoader.ts

```diff
@@ -45,6 +47,148 @@ export class SPLATFileLoader implements ISceneLoaderPluginAsync, ISceneLoaderPlu
         return false;
     }
 
+    /**
+     * Code from https://github.com/dylanebert/gsplat.js/blob/main/src/loaders/PLYLoader.ts Under MIT license
+     * Loads a .ply from data array buffer
+     * if data array buffer is not ply, returns the original buffer
+     */
+    private _loadPLY(data: any): ArrayBuffer {
+        const ubuf = new Uint8Array(data);
+        const header = new TextDecoder().decode(ubuf.slice(0, 1024 * 10));
+        const headerEnd = "end_header\n";
+        const headerEndIndex = header.indexOf(headerEnd);
+        if (headerEndIndex < 0 || !header) {
+            return data;
+        }
+        const vertexCount = parseInt(/element vertex (\d+)\n/.exec(header)![1]);
+
+        let rowOffset = 0;
+        const offsets: Record<string, number> = {
+            double: 8,
+            int: 4,
+            uint: 4,
+            float: 4,
+            short: 2,
+            ushort: 2,
+            uchar: 1,
+        };
+
+        type PlyProperty = {
+            name: string;
+            type: string;
+            offset: number;
+        };
+        const properties: PlyProperty[] = [];
+        for (const prop of header
+            .slice(0, headerEndIndex)
+            .split("\n")
+            .filter((k) => k.startsWith("property "))) {
+            const [_p, type, name] = prop.split(" ");
+            properties.push({ name, type, offset: rowOffset });
+            if (!offsets[type]) throw new Error(`Unsupported property type: ${type}`);
+            rowOffset += offsets[type];
+        }
+
+        const rowLength = 3 * 4 + 3 * 4 + 4 + 4;
+        const SH_C0 = 0.28209479177387814;
+
+        const dataView = new DataView(data, headerEndIndex + headerEnd.length);
+        const buffer = new ArrayBuffer(rowLength * vertexCount);
+        const q = new Quaternion();
+
+        for (let i = 0; i < vertexCount; i++) {
+            const position = new Float32Array(buffer, i * rowLength, 3);
+            const scale = new Float32Array(buffer, i * rowLength + 12, 3);
+            const rgba = new Uint8ClampedArray(buffer, i * rowLength + 24, 4);
+            const rot = new Uint8ClampedArray(buffer, i * rowLength + 28, 4);
+
+            let r0: number = 255;
+            let r1: number = 0;
+            let r2: number = 0;
+            let r3: number = 0;
+
+            properties.forEach((property) => {

```

Comment:
You could use a normal `for` because there's a function call every time the `forEach` loops, and because there can be a lot of vertices, it could make a noticeable difference in the end.

Reply:
Fixed